### PR TITLE
Lack of stop event on system.data.sqlclient.sqlconnection.open breaks EventSource activity tracking #540

### PR DESCRIPTION
**Pull request by iusafaro**
Lack of stop event on system.data.sqlclient.sqlconnection.open breaks EventSource activity tracking #540

### CODE REVIEW

File: Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs [hunk 1]
```diff
@@ -143,6 +143,22 @@ public void OnEndSendsCorrectTelemetry_1ArgumentOverride()
         }
 
         [TestMethod]
```
Comment (SergeyKanzhelev) [inline Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs]: you need a test to validate that `End` for dependency telemetry will in fact be called

File: Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs [hunk 2]
```diff
@@ -402,22 +413,29 @@ private void OnEndInternal(object exceptionObj, object thisObj)
...
+                        telemetry.Success = false;
+                        telemetry.Properties.Add("ErrorMessage", exception.Message);
+
+                        var sqlEx = exception as SqlException;
+                        telemetry.ResultCode = sqlEx != null ? sqlEx.Number.ToString(CultureInfo.InvariantCulture) : "0";
+                    }
+                    else
+                    {
+                        telemetry.Success = true;
+                        telemetry.ResultCode = "0";
```
Comment (SergeyKanzhelev) [inline Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs]: this is not required as `ResultCode` is optional. It's not relevant to this review though...

File: Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs [hunk 3]
```diff
@@ -371,7 +381,8 @@ private void OnEnd(object exceptionObj, object thisObj)
         /// </summary>
         /// <param name="exceptionObj">The exception object if any.</param>
         /// <param name="thisObj">This object.</param>
-        private void OnEndInternal(object exceptionObj, object thisObj)
+        /// <param name="processTelemetryItem">True if telemetry item should be processed.</param>
```
Comment (SergeyKanzhelev) [inline Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs]: name is not clear. Can you call it `sendTelemetryItem` instead. Description for `param` should also be more specific

File: Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs (hunk 1 above)
Comment (iusafaro) [inline Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs]: done. also updated OnEndExceptionAsync to stop operation.

**Pull request merged**